common/source/DynaLog.cpp

```diff
@@ -114,37 +114,43 @@ void Logger::addStream(std::ostream &stream) {
 
 void Logger::trace(std::string file, std::string func, int line_num,
                    const LogContext &context, const std::string &message) {
-  if (m_log_level >= LogLevel::TRACE) {
+  if (static_cast<unsigned int>(m_log_level) >=
+      static_cast<unsigned int>(LogLevel::TRACE)) {
```

Comment:
I could probably leave this out, because I explicitly make the enum an unsigned int when it is constructed. Though I'm as you might have noticed a fan of being explicit sometimes at the detriment of clarity. If you think it makes sense to remove it we can.